File: sdks/dotnet/src/Models/StockTransfer.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Puxbay.SDK.Models
{
    public class StockTransfer
    {
        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("source_branch")]
        public string SourceBranch { get; set; }

        [JsonProperty("source_branch_name")]
        public string SourceBranchName { get; set; }

        [JsonProperty("destination_branch")]
        public string DestinationBranch { get; set; }

        [JsonProperty("destination_branch_name")]
        public string DestinationBranchName { get; set; }

        [JsonProperty("status")]
        public string Status { get; set; }

        [JsonProperty("items")]
        public List<Dictionary<string, object>> Items { get; set; }

        [JsonProperty("notes")]
        public string Notes { get; set; }

        [JsonProperty("created_at")]
        public DateTime? CreatedAt { get; set; }

        [JsonProperty("updated_at")]
        public DateTime? UpdatedAt { get; set; }
    }
}

File: sdks/php/src/Models/StockTransfer.php
<?php

namespace Puxbay\Models;

class StockTransfer
{
    public function __construct(
        public readonly string $id,
        public readonly string $sourceBranch,
        public readonly ?string $sourceBranchName = null,
        public readonly string $destinationBranch,
        public readonly ?string $destinationBranchName = null,
        public readonly string $status,
        public readonly array $items,
        public readonly ?string $notes = null,
        public readonly \DateTimeImmutable $createdAt,
        public readonly \DateTimeImmutable $updatedAt,
    ) {
    }

    public static function fromArray(array $data): self
    {
        return new self(
            id: $data['id'],
            sourceBranch: $data['source_branch'],
            sourceBranchName: $data['source_branch_name'] ?? null,
            destinationBranch: $data['destination_branch'],
            destinationBranchName: $data['destination_branch_name'] ?? null,
            status: $data['status'],
            items: $data['items'] ?? [],
            notes: $data['notes'] ?? null,
            createdAt: new \DateTimeImmutable($data['created_at']),
            updatedAt: new \DateTimeImmutable($data['updated_at']),
        );
    }

    public function toArray(): array
    {
        return [
            'id' => $this->id,
            'source_branch' => $this->sourceBranch,
            'destination_branch' => $this->destinationBranch,
            'status' => $this->status,
            'items' => $this->items,
            'notes' => $this->notes,
        ];
    }
}
